### ISSUE TITLE
NoClassDefFoundError: SubnetUtils

### ISSUE THREAD

**Issue by syepes** [prestodb/presto#5124]
I am getting the following error when trying to query a Hive table using ether 0.145 or 0.141t
Is anyone else also facing this issue?

JDK v1.8.0_72-b15
```
2016-04-26T22:20:24.414+0200    WARN    hive-hive-110   com.facebook.presto.hive.util.ResumableTasks    ResumableTask completed exceptionally
java.lang.NoClassDefFoundError: com/facebook/presto/hadoop/$internal/org/apache/commons/net/util/SubnetUtils
        at org.apache.hadoop.net.NetUtils.isValidSubnet(NetUtils.java:838)
        at org.apache.hadoop.hdfs.DFSClient.getLocalInterfaceAddrs(DFSClient.java:741)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:686)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:619)
        at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:149)
        at org.apache.hadoop.fs.PrestoFileSystemCache.createFileSystem(PrestoFileSystemCache.java:74)
        at org.apache.hadoop.fs.PrestoFileSystemCache.getInternal(PrestoFileSystemCache.java:61)
        at org.apache.hadoop.fs.PrestoFileSystemCache.get(PrestoFileSystemCache.java:43)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:371)
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
        at com.facebook.presto.hive.HdfsEnvironment.getFileSystem(HdfsEnvironment.java:53)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader.loadPartition(BackgroundHiveSplitLoader.java:320)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader.loadSplits(BackgroundHiveSplitLoader.java:232)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader.access$500(BackgroundHiveSplitLoader.java:71)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.lambda$process$0(BackgroundHiveSplitLoader.java:188)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:422)
        at org.apache.hadoop.security.UserGroupInformation$3.run(UserGroupInformation.java:1677)
        at org.apache.hadoop.security.AsSubjectInvoker.doAs(AsSubjectInvoker.java:49)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1673)
        at com.facebook.presto.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.process(BackgroundHiveSplitLoader.java:187)
        at com.facebook.presto.hive.util.ResumableTasks.safeProcessTask(ResumableTasks.java:45)
        at com.facebook.presto.hive.util.ResumableTasks.lambda$submit$1(ResumableTasks.java:33)
        at io.airlift.concurrent.BoundedExecutor.executeOrMerge(BoundedExecutor.java:69)
        at io.airlift.concurrent.BoundedExecutor.access$000(BoundedExecutor.java:28)
        at io.airlift.concurrent.BoundedExecutor$1.run(BoundedExecutor.java:40)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by martint** [comment]
We probably just need to fix the shading rules in presto-hadoop-* to not exclude that class.

@syepes , which of the hive connectors are you using?

**Comment by syepes** [comment]
@martint I am using the connector.name=hive-hadoop2

**Comment by electrum** [comment]
Fixed in 0.147
